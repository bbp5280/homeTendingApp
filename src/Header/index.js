import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Header extends Component {
  constructor(){
    super();
    this.state = {
      showLogIn: true
    };
  }
  handleShowLogIn(){
    this.setState({
      showLogIn:!this.state.showLogIn
    });
  }

  render(){
    console.log(this.state);
    return (
      <header className='header'>
        <div>
          <h1>HOME TENDING</h1>
          <div className='log-in-div'>
            { this.state.showLogIn &&
            <Link to='/admin/messages' className='log-in' onClick={() => this.handleShowLogIn()}>Log In </Link>
          }
          </div>
        </div>
      </header>
    );
  }
}

export default Header;
