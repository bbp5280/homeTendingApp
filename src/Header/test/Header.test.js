import { shallow, configure } from 'enzyme';
import  Header from '../index';
import React from 'react';
import  Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

describe('CardCatalog snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<Header/>);

    expect(wrapper).toMatchSnapshot();

  });
});
