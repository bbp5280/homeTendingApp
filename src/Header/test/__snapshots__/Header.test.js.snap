exports[`CardCatalog snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1>
                HOME TENDING
        </h1>
        <div
                className="log-in-div">
                <Link
                        className="log-in"
                        replace={false}
                        to="/admin/messages">
                        Log In 
                </Link>
        </div>
</div>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            HOME TENDING
</h1>,
          <div
            className="log-in-div">
            <Link
                        className="log-in"
                        replace={false}
                        to="/admin/messages">
                        Log In 
            </Link>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "HOME TENDING",
          },
          "ref": null,
          "rendered": "HOME TENDING",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="log-in"
              replace={false}
              to="/admin/messages">
              Log In 
</Link>,
            "className": "log-in-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Log In ",
              "className": "log-in",
              "replace": false,
              "to": "/admin/messages",
            },
            "ref": null,
            "rendered": "Log In ",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
                    HOME TENDING
          </h1>
          <div
                    className="log-in-div">
                    <Link
                              className="log-in"
                              replace={false}
                              to="/admin/messages">
                              Log In 
                    </Link>
          </div>
</div>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              HOME TENDING
</h1>,
            <div
              className="log-in-div">
              <Link
                            className="log-in"
                            replace={false}
                            to="/admin/messages">
                            Log In 
              </Link>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "HOME TENDING",
            },
            "ref": null,
            "rendered": "HOME TENDING",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="log-in"
                replace={false}
                to="/admin/messages">
                Log In 
</Link>,
              "className": "log-in-div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Log In ",
                "className": "log-in",
                "replace": false,
                "to": "/admin/messages",
              },
              "ref": null,
              "rendered": "Log In ",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
